src/schedule/ll_schedule.c

```diff
@@ -167,6 +167,10 @@ static void schedule_ll_client_reschedule(struct ll_schedule_data *sch)
 		list_for_item(tlist, &sch->tasks) {
 			task = container_of(tlist, struct task, list);
 
+			/* only check tasks asked for rescheduling */
+			if (task->state != SOF_TASK_STATE_RESCHEDULE)
```

Comment:
shouldn't `QUEUED` be considered too?

Reply:
No, the schedule_ll_client_reschedule() is used for re-scheduling those tasks have been scheduled and asking for rescheduling only. So the state machine like this:
INIT--(.schedule_task())-->QUEUED--(_is_pending())-->PENDING--(_execute())-->RUNNING--(task_run() done)-->RESCHEDULE--(_client_reschedule() to update timer, _is_pending())-->PENDING.
```



                 ┌─────┐
        ┌────────┤INIT │
        │        └─────┘
        │
        |
        │
     ┌──▼───┐   ┌───────┐   ┌──────────┐
     │QUEUED├───►PENDING◄───┤RESCHEDUL │
     └──────┘   └───┬───┘   └────▲─────┘
                    │            |                   
                    │            |
                    │            │
                ┌───▼───┐        |
                |       |        |
                │RUNNING├────────┘
                └───┬───┘
                    │
               ┌────▼────┐
               │COMPLETED│
               └─────────┘       


```

Reply:
Would be nice to add this ASCII art in the code comments.